### PR TITLE
fix(types): add vendors

### PR DESCRIPTION
**Pull request by smalexan**
## Description of this change

> Add vendor schema

## Why is this change being made?
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have evaluated the security impact of this change, and [OWASP Secure Coding Practices](https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/migrated_content#) have been observed.
- [x] I have informed stakeholders of my changes.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -106,26 +106,26 @@ export interface GetConsentRequest {
...
   applicationEnvironmentCode: string;
-  identities: string[];
-  identityMap: Map<string,string>
+  identities: {[key: string]: string}
   purposes: {[key: string]: PurposeLegalBasis};
 }
 
 export interface GetConsentResponse {
   purposes: {[key: string]: PurposeAllowed};
+  vendors: string[]; // list of vendor ids for which the user has opted out
```
Comment (smalexan) [inline src/index.ts:115]: does naming make sense for opt out? this matches wheelhouse

File: src/index.ts [hunk 2]
```diff
@@ -318,6 +317,35 @@ export interface Theme {
...
+  description: string;
+  descriptionLegal: string;
+}
+
+export interface GVL {
+  purposes: {[key: string]:  GVLPurpose};
+  specialPurposes: {[key: string]:  GVLPurpose};
+  features: {[key: string]:  GVLPurpose};
+  specialFeatures: {[key: string]:  GVLPurpose};
+}
```
Comment (smalexan) [inline src/index.ts:332]: leveraged for https://www.figma.com/file/rr4W8wjE0HwtgvI6j52RVY/Experience-Server?node-id=2518%3A8643. All vendor purposes listed even if customer only supports a subset

Comment (sethyates) [reply src/index.ts:332]: One thing to note is as per TCF policies, we must represent their GVL "without modifications" (filtering allowed).

File: src/index.ts [hunk 3]
```diff
@@ -318,25 +317,56 @@ export interface Theme {
...
+
+export interface VendorPurpose {
+  id: string;
+  legalBasisCode?: string;
```
Comment (sethyates) [inline src/index.ts:336]: Is this really a thing in TCF? Is this different to our legalBasisCodes?

File: src/index.ts [hunk 4]
```diff
@@ -370,7 +400,8 @@ export function getBootstrapConfiguration(request: GetBootstrapConfigurationRequ
 
 // Gets the full configuration for the specified parameters.
 export function getFullConfiguration(request: GetFullConfigurationRequest): Promise<Configuration> {
-  const url = `/config/${request.organizationCode}/${request.appCode}/${request.envCode}/${request.hash}/${request.policyScopeCode}/${request.languageCode}/config.json`;
+  const url = `/config/${request.organizationCode}/${request.appCode}/${request.envCode}/${request.hash}/
+  ${request.policyScopeCode}/${request.languageCode}/config.json`;
```
Comment (sethyates) [inline src/index.ts:404]: So this got wrapped. Probably won't work?

**Review by sethyates — APPROVED** [review]

**Pull request merged**